cmake-naming.md

```diff
@@ -0,0 +1,36 @@
+## CMake targets for Mbed "targets"
+
+### targets.json naming scheme
+
+- `MCU_STM32L4` - base target
+- `DISCO_L4R9I` - final target - a user board or could be custom target, inherits from MCU_  target
+
+The targets.json properties use capital letters with underscore but not the our config. Probably to distinguish these from other regular directories (the initial Mbed SDK had it this way).
+
+### CMake naming scheme
+
+All Mbed OS CMake targets are prefixed with `mbed-` to avoid clashes with user's targets (out of the tree targets): `mbed-ble`, `mbed-nanostack`.
```

Comment:
> (out of the tree targets)

I am not sure what this means.

> : `mbed-ble`, `mbed-nanostack`.

Why are these CMake targets listed?

Reply:
custom targets (out of Mbed OS code base)

mbed-ble is as an example for naming we have currently